cumulusci/utils/git.py

```diff
@@ -6,14 +6,17 @@ def git_path(repo_root, tail=None):
     with tail appended (if present) or None if repo_root is not set.
     """
     path = None
-    if repo_root:
+    if repo_root and (pathlib.Path(repo_root) / ".git").exists():
+        # When we're a source tarball, we'll have a repo_root but no .git/HEAD.
```

Comment:
What was the scenario where we were ending up with a source tarball? CumulusCI currently generally requires a project to be a git repository, and I think changing that would be desirable but involve more than just this.

Reply:
@davisagli I was observing this behavior when running `create_package_version` as a cross-project task - it failed in the `source`'s working dir while trying to get the current branch. (This seemed like an expedient way to fix it but happy for any better recommendations!)

Reply:
For cross-project tasks, the github source attempts to construct a project config which has repository details preconfigured, so that looking for .git isn't needed: https://github.com/SFDO-Tooling/CumulusCI/blob/master/cumulusci/core/source/github.py#L117

It's not setting the branch though. Can you give me steps to reproduce this, or a traceback? I'd like to see what is looking for the branch.

Also, what's the use case for running create_package_version as a cross-project task?

Reply:
The use case and the repro - Elevate. https://github.com/SalesforceFoundation/spc-integration/pull/291/files

This PR builds the Transport Layer from a feature branch as a 2GP, and also builds and installs the NPSP Connector as a 2GP (cross-project) to facilitate the team doing end-to-end testing of feature changes, which is currently very difficult. NPSP Connector has a dependency on TL.

Reply:
@davidmreed I think I'm going to fix this by making create_package_version not set the branch when creating the Package2CreateVersionRequest, because I don't think it actually does anything for us. I think changing `git_path` to return None instead of raising when there's no .git folder will require a more careful analysis to make sure we aren't introducing NPE-style errors elsewhere.